diff --git a/Core/Unix/Linux/CoreLinux.cpp b/Core/Unix/Linux/CoreLinux.cpp
index 634a3a2..a4837d4 100644
--- a/Core/Unix/Linux/CoreLinux.cpp
+++ b/Core/Unix/Linux/CoreLinux.cpp
@@ -39,6 +39,15 @@ namespace TrueCrypt
 		loopPaths.push_back ("/dev/loop/");
 		loopPaths.push_back ("/dev/.static/dev/loop");
 
+		// On some distributions, "losetup -f" must be called first to create a default loop device
+		list <string> args;
+		args.push_back ("-f");
+		try
+		{
+			Process::Execute ("losetup", args);
+		}
+		catch (...) { }
+
 		for (int devIndex = 0; devIndex < 256; devIndex++)
 		{
 			string loopDev;
