diff --git a/Main/GraphicUserInterface.cpp b/Main/GraphicUserInterface.cpp
index 04426bd..eb5fa03 100644
--- a/Main/GraphicUserInterface.cpp
+++ b/Main/GraphicUserInterface.cpp
@@ -991,22 +991,15 @@ namespace TrueCrypt
 		return;
 
 #else
-		wxMimeTypesManager mimeMgr;
-		wxFileType *fileType = mimeMgr.GetFileTypeFromExtension (document.GetExt());
-		if (fileType)
+
+		try
 		{
-			try
-			{
-#ifdef TC_MACOSX
-				if (wxExecute (fileType->GetOpenCommand (document.GetFullPath())) != 0)
-					return;
-#else
-				if (wxExecute (fileType->GetOpenCommand (L"\"" + document.GetFullPath() + L"\"")) != 0)
-					return;
-#endif
-			}
-			catch (TimeOut&) { }
+			list <string> args;
+			args.push_back (string (document.GetFullPath().mb_str()));
+			Process::Execute ("xdg-open", args, 2000);
 		}
+		catch (TimeOut&) { }
+		catch (exception &e) { ShowError (e); }
 #endif
 	}
 
@@ -1085,7 +1078,7 @@ namespace TrueCrypt
 #elif defined (TC_MACOSX)
 			docPath += L"/../Resources/TrueCrypt User Guide.pdf";
 #elif defined (TC_UNIX)
-			docPath = L"/usr/share/truecrypt/doc/TrueCrypt User Guide.pdf";
+			docPath = L"/usr/share/doc/truecrypt/TrueCrypt User Guide.pdf.gz";
 #else
 #	error TC_RESOURCE_DIR undefined
 #endif
